lib/json_schema/validator.rb

```diff
@@ -506,7 +506,13 @@ def validate_type(schema, data, errors, path)
       if valid_types.any? { |t| data.is_a?(t) }
         true
       else
-        message = %{#{data.inspect} is not #{ErrorFormatter.to_list(schema.type)}.}
+        key = if schema.fragment =~ /patternProperties/
```

Comment:
Bear with me as I haven't looked closely at this project in a while, but do we need the regex match here? Shouldn't `fragment` just represent the last piece of the pointer?